log-manager/src/main/java/io/airlift/log/SocketMessageOutput.java

```diff
@@ -0,0 +1,99 @@
+package io.airlift.log;
+
+import com.google.common.net.HostAndPort;
+import org.weakref.jmx.Managed;
+
+import javax.annotation.concurrent.GuardedBy;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.net.InetSocketAddress;
+import java.net.Socket;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.logging.Formatter;
+
+public class SocketMessageOutput
+        implements MessageOutput
+{
+    private static final int CONNECTION_TIMEOUT_MILLIS = 100;
+    private static final int MAX_WRITE_ATTEMPTS_PER_MESSAGE = 5;
+
+    private final InetSocketAddress socketAddress;
+    private final AtomicLong failedConnections = new AtomicLong(0);
+
+    @GuardedBy("this")
+    private Socket socket;
+    @GuardedBy("this")
+    private OutputStream currentOutputStream;
+
+    private SocketMessageOutput(String hostAndPortString)
```

Comment:
Make the parameter type here and `createSocketHandler` be `HostAndPort`, and force the caller to deal with parsing and any errors